pymc_marketing/clv/models/modified_beta_geo.py

```diff
@@ -0,0 +1,316 @@
+"""Modified Beta-Geometric Negative Binomial Distribution (MBG/NBD) model for a non-contractual customer population across continuous time."""  # noqa: E501
+
+import warnings
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+import pytensor.tensor as pt
+import xarray
+from pymc.distributions.dist_math import check_parameters
+from scipy.special import hyp2f1
+
+from pymc_marketing.clv.models import BetaGeoModel
+from pymc_marketing.clv.utils import to_xarray
+
+
+class ModBetaGeoModel(BetaGeoModel):
+    r"""Also known as the MBG/NBD model.
+
+    Based on [5]_, [6]_, this model has the following assumptions:
+    1) Each individual, ``i``, has a hidden ``lambda_i`` and ``p_i`` parameter
+    2) These come from a population wide Gamma and a Beta distribution
+       respectively.
+    3) Individuals purchases follow a Poisson process with rate :math:`\lambda_i*t` .
+    4) At the beginning of their lifetime and after each purchase, an
+       individual has a p_i probability of dieing (never buying again).
+
+    References
+    ----------
+    .. [5] Batislam, E.P., M. Denizel, A. Filiztekin (2007),
+       "Empirical validation and comparison of models for customer base
+       analysis,"
+       International Journal of Research in Marketing, 24 (3), 201-209.
+    .. [6] Wagner, U. and Hoppe D. (2008), "Erratum on the MBG/NBD Model,"
+       International Journal of Research in Marketing, 25 (3), 225-226.
+    """
+
+    _model_type = "MBG/NBD"
+
+    def build_model(self) -> None:  # type: ignore[override]
+        """Build the model."""
+        coords = {"customer_id": self.data["customer_id"]}
+        with pm.Model(coords=coords) as self.model:
+            # purchase rate priors
+            alpha = self.model_config["alpha_prior"].create_variable("alpha")
+            r = self.model_config["r_prior"].create_variable("r")
+
+            # dropout priors
+            if "a_prior" in self.model_config and "b_prior" in self.model_config:
+                a = self.model_config["a_prior"].create_variable("a")
+                b = self.model_config["b_prior"].create_variable("b")
+            else:
+                # hierarchical pooling of dropout rate priors
+                phi_dropout = self.model_config["phi_dropout_prior"].create_variable(
+                    "phi_dropout"
+                )
+                kappa_dropout = self.model_config[
+                    "kappa_dropout_prior"
+                ].create_variable("kappa_dropout")
+
+                a = pm.Deterministic("a", phi_dropout * kappa_dropout)
+                b = pm.Deterministic("b", (1.0 - phi_dropout) * kappa_dropout)
+
+            def logp(t_x, x, a, b, r, alpha, T):
+                """
+                Compute the log-likelihood of the BG/NBD model.
+
+                The log-likelihood expression here aligns with expression (4) from [3]
+                due to the possible numerical instability of expression (3).
```

Comment:
Delete L81-82; they only pertain to `BetaGeoModel`.

Reply:
Removed